zio-http/src/main/scala/zio/http/Server.scala

```diff
@@ -1,300 +1,62 @@
-   */
-  def withError[R1](errorHandler: Throwable => ZIO[R1, Nothing, Unit]): Server[R with R1, E] =
-    Concat(self, Server.Error(errorHandler))
-
-  /**
-   * Creates a new server using netty FlowControlHandler if enable (@see <a
-   * href="https://netty.io/4.1/api/io/netty/handler/flow/FlowControlHandler.html">FlowControlHandler</a>).
-   */
-  def withFlowControl(enable: Boolean): Server[R, E] = Concat(self, Server.FlowControl(enable))
-
-  /**
-   * Creates a new server with netty's HttpServerKeepAliveHandler to close
-   * persistent connections when enable is true (@see <a
-   * href="https://netty.io/4.1/api/io/netty/handler/codec/http/HttpServerKeepAliveHandler.html">HttpServerKeepAliveHandler</a>).
-   */
-  def withKeepAlive(enable: Boolean): Server[R, E] = Concat(self, KeepAlive(enable))
-
-  /**
-   * Creates a new server with the leak detection level provided (@see <a
-   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html">ResourceLeakDetector.Level</a>).
-   */
-  def withLeakDetection(level: LeakDetectionLevel): Server[R, E] = Concat(self, LeakDetection(level))
-
-  /**
-   * Creates a new server with HttpObjectAggregator with the specified max size
-   * of the aggregated content.
-   */
-  def withObjectAggregator(maxRequestSize: Int = 1024 * 100): Server[R, E] =
-    Concat(self, ObjectAggregator(maxRequestSize))
-
-  /**
-   * Creates a new server listening on the provided port.
-   */
-  def withPort(port: Int): Server[R, E] = Concat(self, Server.Address(new InetSocketAddress(port)))
-
-  /**
-   * Creates a new server with netty's HttpContentDecompressor to decompress
-   * Http requests (@see <a href =
-   * "https://netty.io/4.1/api/io/netty/handler/codec/http/HttpContentDecompressor.html">HttpContentDecompressor</a>).
-   */
-  def withRequestDecompression(enabled: Boolean, strict: Boolean): Server[R, E] =
-    Concat(self, RequestDecompression(enabled, strict))
-
-  /**
-   * Creates a new server with the following ssl options.
-   */
-  def withSsl(sslOptions: ServerSSLOptions): Server[R, E] = Concat(self, Server.Ssl(sslOptions))
-
-  /**
-   * Creates a new server by passing a function that modifies the channel
-   * pipeline. This is generally not required as most of the features are
-   * directly supported, however think of this as an escape hatch for more
-   * advanced configurations that are not yet support by ZIO Http.
-   *
-   * NOTE: This method might be dropped in the future.
-   */
-  def withUnsafeChannelPipeline(unsafePipeline: ChannelPipeline => Unit): Server[R, E] =
-    Concat(self, UnsafeChannelPipeline(unsafePipeline))
-
-  /**
-   * Provides unsafe access to netty's ServerBootstrap. Modifying server
-   * bootstrap is generally not advised unless you know what you are doing.
-   */
-  def withUnsafeServerBootstrap(unsafeServerbootstrap: ServerBootstrap => Unit): Server[R, E] =
-    Concat(self, UnsafeServerBootstrap(unsafeServerbootstrap))
+  def port: UIO[Int]
```

Comment:
If the port never changes for a running server, this could be `def port: Int` instead.

Reply:
if the configuration is having port set to 0, then it means that netty will choose first free port on the machine. So it is dynamic, that's I added it in this way. However, it become problematic to obtain the binding port with the latest changes.